app/src/main/java/com/github/geohunt/app/ui/components/bounties/AdminBountyPage.kt

```diff
@@ -0,0 +1,248 @@
+import com.github.geohunt.app.i18n.DateFormatUtils
+import com.github.geohunt.app.i18n.toSuffixedString
+import com.github.geohunt.app.model.Bounty
+import com.github.geohunt.app.model.Challenge
+import com.github.geohunt.app.model.Team
+import com.github.geohunt.app.ui.components.bounties.viewmodel.AdminBountyViewModel
+import com.github.geohunt.app.ui.screens.home.HorizontalDivider
+import java.time.LocalDateTime
+
+@Composable
+fun AdminBountyPage(
+    bid: String,
+    onFailure: (Throwable) -> Unit = {},
+    onCreateChallenge: () -> Unit = {},
+    viewModel: AdminBountyViewModel = viewModel(factory = AdminBountyViewModel.Factory)
+) {
+    val bountyState = viewModel.bounty.collectAsState()
+
+    LaunchedEffect(bid, viewModel) {
+        viewModel.withBusinessId(bid, onFailure)
+    }
+
+    if (bountyState.value != null) {
+        val teamsState = viewModel.teamsFlow!!.collectAsState(initial = listOf())
+        val challengesState = viewModel.challenges.collectAsState()
+
+        AdminBountyPageUI(
+            bounty = bountyState.value!!,
+            teams = teamsState.value,
+            challenges = challengesState.value,
+            setName = viewModel::setBountyName,
+            onCreateChallenge = onCreateChallenge
+        )
+    } else {
+        CircularProgressIndicator()
+    }
+}
+
+@Composable
+private fun AdminBountyPageUI(
+    bounty: Bounty,
+    setName: (String) -> Unit,
+    teams: List<Team>,
+    challenges: List<Challenge>,
+    onCreateChallenge: () -> Unit,
+) {
+    val memberCount = teams.sumOf { it.membersUid.size }
+    var showPopup by remember { mutableStateOf(false) }
+
+    if (showPopup) {
+        RenameBountyPopup(bounty) { newNameOpt ->
+            showPopup = false
+            newNameOpt?.apply { setName(this) }
+        }
+    }
+
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .verticalScroll(rememberScrollState())
+            .padding(5.dp, 10.dp)
+            .testTag("admin-bounty-page-loaded")
+    ) {
+        Row(modifier = Modifier.padding(10.dp, 0.dp)) {
+            Text(
+                text = bounty.name,
+                fontSize = 28.sp
+            )
+
+            Spacer(modifier = Modifier.weight(1f))
+
+            IconButton(onClick = { showPopup = true },
+                modifier = Modifier
+                    .testTag("edit-btn")
+                    .align(Alignment.CenterVertically)) {
+                Icon(Icons.Default.Edit,
+                    contentDescription = "edit")
+            }
+
+            Spacer(modifier = Modifier.width(5.dp))
+
+            Icon(Icons.Default.AdminPanelSettings,
+                modifier = Modifier.align(Alignment.CenterVertically),
+                contentDescription = "admin"
+            )
+        }
+
+        Row(modifier = Modifier.padding(20.dp, 0.dp)) {
+            val firstComponent = formatTimeStuff(start = bounty.startingDate, end = bounty.expirationDate)

```

Comment:
Please use better names

Reply:
AHAHAHA I forgot about this one

Reply:
To my defense this is a private function